test/Microsoft.AspNet.Routing.Tests/Constraints/HttpMethodRouteConstraintTests.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Http.Internal;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNet.Routing.Constraints
+{
+    public class HttpMethodRouteConstraintTests
+    {
+        [Theory]
+        [InlineData("GET")]
+        [InlineData("PosT")]
+        public void HttpMethodRouteConstraint_UrlMatching_AcceptsAllowedMethods(string httpMethod)
+        {
+            // Arrange
+            var constraint = new HttpMethodConstraint("GET", "post");
+
+            var httpContext = new DefaultHttpContext();
+            httpContext.Request.Method = httpMethod;
+            var route = Mock.Of<IRouter>();
+
+            var values = new RouteValueDictionary(new { httpMethod = "IGNORED" });
+
+            // Act
+            var result = constraint.Match(httpContext, route, "httpMethod", values, RouteDirection.IncomingRequest);
+
+            // Assert
+            Assert.True(result);
+        }
+
+        [Theory]
+        [InlineData("OPTIONS")]
+        [InlineData("SomeRandomThing")]
+        public void HttpMethodRouteConstraint_UrlMatching_RejectsOtherMethods(string httpMethod)
+        {
+            // Arrange
+            var constraint = new HttpMethodConstraint("GET", "post");
+
+            var httpContext = new DefaultHttpContext();
+            httpContext.Request.Method = httpMethod;
+            var route = Mock.Of<IRouter>();
+
+            var values = new RouteValueDictionary(new { httpMethod });
+
+            // Act
+            var result = constraint.Match(httpContext, route, "httpMethod", values, RouteDirection.UrlGeneration);
+
+            // Assert
+            Assert.False(result);
+        }
+
+        [Theory]
+        [InlineData("GET")]
+        [InlineData("PosT")]
+        public void HttpMethodRouteConstraint_UrlGeneration_AcceptsAllowedMethods(string httpMethod)
+        {
+            // Arrange
+            var constraint = new HttpMethodConstraint("GET", "post");
+
+            var httpContext = new DefaultHttpContext();
+            httpContext.Request.Method = "IGNORED";
```

Comment:
It can, I felt like this was good to have, but I guess it's an example of testing that we don't have a feature that we don't have :laughing:. Something I usually tell other not to do.